utility.py

```diff
@@ -56,6 +56,7 @@ def main():
 def write_file(data, filename):
     ''' write bytes to file '''
     print("  writing %s bytes to %s" %(len(data), filename))
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
```

Comment:
This API requires Python 3.2.  I'm not sure if you have a minimum supported version, or if this is ok?  There are other solutions using a try/except block if we need to support < 3.2